src/Controllers/User.php

```diff
@@ -6,82 +6,131 @@
 
 use App\Model\DatabaseConnection;
 use App\Repository\UserRepository;
+use Twig\Environment;
+use Twig\Error\LoaderError;
+use Twig\Error\RuntimeError;
+use Twig\Error\SyntaxError;
 
-class User extends Homepage
+class User
 {
+    public function __construct(public Environment $twig)
+    {
+        $this->twig->addGlobal('session', $_SESSION);
+    }
+
+    /**Connect user and redirect to homepage
+     *
+     * @throws SyntaxError
+     * @throws RuntimeError
+     * @throws LoaderError
+     */
     public function signUpAction(): void
     {
         if (count($_POST) > 0) {
-            $pseudo = $_POST['pseudo'];
-            $mail = $_POST['mail'];
-            $password = $_POST['password'];
+            if (isset($_POST['pseudo']) && !empty($_POST['pseudo'])
+                && isset($_POST['mail']) && !empty($_POST['mail'])
+                && isset($_POST['password']) && !empty($_POST['password'])) {
+                $pseudo = filter_input(INPUT_POST, 'pseudo',FILTER_SANITIZE_SPECIAL_CHARS);
+                $mail = filter_input(INPUT_POST, 'mail',FILTER_SANITIZE_EMAIL);
+                $password = filter_input(INPUT_POST, 'password',FILTER_SANITIZE_SPECIAL_CHARS);
 
-            $signUp = new UserRepository();
-            $signUp->connection = new DatabaseConnection();
+                $signUp = new UserRepository();
+                $signUp->connection = new DatabaseConnection();
+                $signUpHomepage = new Homepage($this->twig);
```

Comment:
variable $signUpHomepage is probably undefined.
From PHPstorm weak warning, is that a problem ?